src/interface/orthnull.jl

```diff
@@ -1,220 +1,429 @@
+    alg = TruncatedAlgorithm(alg_svd, select_null_truncation(trunc))
+    return LeftNullViaSVD(alg)
+end
+
+function select_algorithm(::typeof(right_null!), A, ::Val{:lq}; trunc = nothing, kwargs...)
+    isnothing(trunc) ||
+        throw(ArgumentError("LQ-based `right_null` is incompatible with specifying `trunc`"))
+    alg = select_algorithm(lq_null!, A, get(kwargs, :lq, nothing))
+    return RightNullViaLQ(alg)
+end
+function select_algorithm(::typeof(right_null!), A, ::Val{:svd}; trunc = nothing, kwargs...)
+    alg_svd = select_algorithm(svd_full!, A, get(kwargs, :svd, nothing))
+    alg = TruncatedAlgorithm(alg_svd, select_null_truncation(trunc))
+    return RightNullViaSVD(alg)
+end
+
+default_algorithm(::typeof(left_orth!), A::TA; trunc = nothing, kwargs...) where {TA} =
+    isnothing(trunc) ? select_algorithm(left_orth!, A, Val(:qr); qr = kwargs) :
+    select_algorithm(left_orth!, A, Val(:svd); svd = kwargs)
+# disambiguate
+default_algorithm(::typeof(left_orth!), ::Type{A}; trunc = nothing, kwargs...) where {A} =
+    isnothing(trunc) ? select_algorithm(left_orth!, A, Val(:qr); qr = kwargs) :
+    select_algorithm(left_orth!, A, Val(:svd); svd = kwargs)
+
+default_algorithm(::typeof(right_orth!), A::TA; trunc = nothing, kwargs...) where {TA} =
+    isnothing(trunc) ? select_algorithm(right_orth!, A, Val(:lq); lq = kwargs) :
+    select_algorithm(right_orth!, A, Val(:svd); svd = kwargs)
+# disambiguate
+default_algorithm(::typeof(right_orth!), ::Type{A}; trunc = nothing, kwargs...) where {A} =
+    isnothing(trunc) ? select_algorithm(right_orth!, A, Val(:lq); lq = kwargs) :
+    select_algorithm(right_orth!, A, Val(:svd); svd = kwargs)
+
+default_algorithm(::typeof(left_null!), A::TA; trunc = nothing, kwargs...) where {TA} =
+    isnothing(trunc) ? select_algorithm(left_null!, A, Val(:qr); qr = kwargs) :
+    select_algorithm(left_null!, A, Val(:svd); svd = kwargs, trunc)
+# disambiguate
+default_algorithm(::typeof(left_null!), ::Type{A}; trunc = nothing, kwargs...) where {A} =
+    isnothing(trunc) ? select_algorithm(left_null!, A, Val(:qr); qr = kwargs) :
+    select_algorithm(left_null!, A, Val(:svd); svd = kwargs, trunc)
+
+default_algorithm(::typeof(right_null!), A::TA; trunc = nothing, kwargs...) where {TA} =
+    isnothing(trunc) ? select_algorithm(right_null!, A, Val(:lq); lq = kwargs) :
+    select_algorithm(right_null!, A, Val(:svd); svd = kwargs, trunc)
+# disambiguate
+default_algorithm(::typeof(right_null!), ::Type{A}; trunc = nothing, kwargs...) where {A} =
+    isnothing(trunc) ? select_algorithm(right_null!, A, Val(:lq); lq = kwargs) :
+    select_algorithm(right_null!, A, Val(:svd); svd = kwargs, trunc)
+
+left_orth_alg(alg::AbstractAlgorithm) = LeftOrthAlgorithm(alg)
+right_orth_alg(alg::AbstractAlgorithm) = RightOrthAlgorithm(alg)
+left_null_alg(alg::AbstractAlgorithm) = LeftNullAlgorithm(alg)
+right_null_alg(alg::AbstractAlgorithm) = RightNullAlgorithm(alg)
```

Comment:
Are these used anywhere?

Reply:
yes, these are (as per your suggestion from above) the functions that map the algorithms to their wrapped counterpart which includes the kind of decomposition.

See eg https://github.com/QuantumKitHub/MatrixAlgebraKit.jl/blob/2cb1ae5b928e8468f2d2d6f7eb8c3775d41917ad/src/implementations/orthnull.jl#L83

In principle we could replace them with the constructors directly, but usually it doesn't hurt to have a layer that is not the constructor inbetween to avoid being stuck with the `Constructor(x) isa Constructor` restriction